promql/engine.go

```diff
@@ -452,13 +406,22 @@ func (ng *Engine) execEvalStmt(ctx context.Context, query *query, s *EvalStmt) (
 			return nil, err
 		}
 
+		// Turn matrix and vector types with protected metrics into
+		// model.* types.
+		switch v := val.(type) {
+		case vector:
+			val = v.value()
```

Comment:
For my understanding: this is now an extra copy of all result metrics that we didn't do before, right?